### DISCUSSION TITLE
Testing a preview url

### DISCUSSION THREAD

**Discussion by zeekrey** [vercel/vercel#4898] [Ideas] [answered]
Hey there,

I currently run local E2E tests with cypress before deployment. But since vercel allows us to deploy every single commit, I would like to run those E2E tests on a deployed state rather than locally. It would be nice if this is well integrated in the whole Github pipeline with its [status checks](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-status-checks). 

Does anybody know if something like this exists? If yes, what are the experiences with this approach?

**Comment by styfle** [comment, marked as answer]
Yes! We use https://www.checklyhq.com to run E2E tests against https://vercel.com preview deployments 👍

**Comment by Vadorequest** [comment]
@zeekrey We built our own Github Actions pipeline (workflow), see https://github.com/UnlyEd/next-right-now/tree/v2-mst-aptd-at-lcz-sty/.github/workflows

It works smoothly, as you can see in action at:
- https://github.com/UnlyEd/next-right-now/pull/68#issuecomment-711161148 (GitHub comment)
- https://github.com/UnlyEd/next-right-now/runs/1271096985 (logs) 

Also, it saves screenshots/videos as Github artifacts, when E2E fail.

@styfle Sometimes, our E2E tests encounter an edge case with Vercel, which is described here (race condition where Cypress runs against the Vercel authentication page, instead of the deployment url): https://github.com/UnlyEd/next-right-now/issues/184

I wonder if you ever encountered this issue and if you have any feedback/advices on this. Thank you!
